apps/system/js/app_usage_metrics.js

```diff
@@ -613,19 +592,77 @@
     };
 
     var deviceInfoQuery = {
-      'deviceinfo.update_channel': 'unknown',
+      'app.update.channel': 'unknown',
+      'developer.menu.enabled': false, // If true, data is probably an outlier
+      'deviceinfo.hardware': 'unknown',
+      'deviceinfo.os': 'unknown',
       'deviceinfo.platform_version': 'unknown',
       'deviceinfo.platform_build_id': 'unknown',
-      'developer.menu.enabled': false // If true, data is probably an outlier
+      'deviceinfo.product_model': 'unknown',
+      'deviceinfo.software': 'unknown',
+      'deviceinfo.update_channel': 'unknown'
     };
 
     // Query the settings db to get some more device-specific information
-    AUM.getSettings(deviceInfoQuery, function(deviceinfo) {
-      data.deviceinfo = deviceinfo;
+    AUM.getSettings(deviceInfoQuery, function(deviceInfo) {
+      data.deviceinfo = deviceInfo;
+      data.simInfo = getSIMInfo();
+
       // Now transmit the data
       send(data);
     });
 
+    function getSIMInfo() {
```

Comment:
I like the idea, but the format of the return object here is somewhat specific (meant to be sent to a server as part of a JSON payload). Ideally though we could share the code between this and ftu_ping.js ..